pkg/backup/backup_compaction.go

```diff
@@ -47,6 +51,76 @@ import (
 	"github.com/gogo/protobuf/types"
 )
 
+var (
+	backupCompactionThreshold = settings.RegisterIntSetting(
+		settings.ApplicationLevel,
+		"backup.compaction.threshold",
+		"the required backup chain length for compaction to be triggered (0 to disable compactions)",
+		0,
+		settings.WithVisibility(settings.Reserved),
+		settings.IntInRangeOrZeroDisable(3, math.MaxInt64),
+	)
+)
+
+// maybeStartCompactionJob will initiate a compaction job off of a triggering
+// incremental job if the backup chain length exceeds the threshold.
+// backupStmt should be the original backup statement that triggered the job.
+func maybeStartCompactionJob(
+	ctx context.Context,
+	execCfg *sql.ExecutorConfig,
+	user username.SQLUsername,
+	triggerJob jobspb.BackupDetails,
+	backupStmt string,
+) (jobspb.JobID, error) {
+	threshold := backupCompactionThreshold.Get(&execCfg.Settings.SV)
```

Comment:
could this block of code get pushed into `maybeTriggerBackupCompaction`? I'd like to avoid that extra read of the backup schedule on all successful backup jobs.  Also, moving this block of code creates two filters for compaction which i think makes things easier to read: the first filter is whether the schedule has any chance of doing compaction, the second filter here is if the chain size meets the threshold, then we can compact. 

instead of `maybeTriggerBackupCompaction and `maybeStartCompactionJob` which sound quite similar. Could the ladder be called `maybeStartCompactionOnChainSize`

Reply:
Or maybe we just move `maybeTriggerBackupCompaction` into `maybeStartCompactionJob`

Reply:
yeah, that would work too lol